### ISSUE TITLE
Progress reporting interferes with minibuffer prompts when using rust-analyzer

### ISSUE THREAD

**Issue by tspiteri** [emacs-lsp/lsp-mode#1714]
Let's say I want to save-some-buffers with <kbd>C-x s</kbd> when using Rust and rust-analyzer. After saving a file, rust-analyzer will run `cargo check` and it will send some progress notifications ``"Running `cargo check`"``. These are displayed by lsp--on-progress, which hides the minibuffer prompt.

It can be very confusing to be in the middle of save-some-buffers without any visual feedback, and the visual feedback is now gone because of the progress notifications. I found this especially confusing when save-some-buffers is started indirectly, for example by an <kbd>M-x compile</kbd>.

I don't think the progress should be displayed in the echo area while the mini buffer is in use. I don't know if this is an issue in lsp--on-progress or in the progress-reporter functions in Emacs. I *think* the Emacs progress-reporter is not meant for background progress reporting: (quoting from the manual)
> When an operation can take a while to finish, you should inform the user about the progress it makes. This way the user can estimate remaining time and clearly see that Emacs is busy working, not hung. A convenient way to do this is to use a progress reporter.

As a workaround I had to block lsp--on-progress from doing anything.
```elisp
(advice-add 'lsp--on-progress :around
            (lambda (&rest r)))
```
First I tried to still run the function if the minibuffer is not active:
```elisp
(advice-add 'lsp--on-progress :around
            (lambda (&rest r)
	      (unless (active-minibuffer-window)
		(apply r))))
```
But unfortunately prompts like that used for save-some-buffers and that used for isearch cannot be detected with active-minibuffer-window. Since I do not know how to detect them, I had to disable lsp--on-progress completely.

**Comment by yyoncho** [comment]
see https://github.com/emacs-lsp/lsp-mode/pull/1707

**Comment by Dushistov** [comment]
I wonder would be possible to use status bar instead of minibuffer for such kind of message?

**Comment by yyoncho** [comment]
fixed with #1707
